lib/datadog/core/configuration/option_definition.rb

```diff
@@ -78,8 +86,13 @@ def helper(name, *_args, &block)
             @helpers[name] = block
           end
 
-          def lazy(value = true)
-            @lazy = value
+          def lazy(_value = true)
+            Datadog::Core.log_deprecation do
+              'The `lazy` option is deprecated. You no longer need to specify an option to be lazy '\
+              "when the default value is a block.\n"\
+              'In the case edge case that you need the default value to be a block, please use the the expriemntal option:'\
+              'experimental_default_proc'
+            end
```

Comment:
Two notes on this:

* "You no longer need to" should perhaps be "you no longer can" -- this option is gone, it's not that you wouldn't ever need it
* Is it worth telling people to use the experimental option? Should we instead tell them that we don't support it all until we figure out what we should do?